tests/functional/CachePoolTest.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+namespace WpOop\TransientCache\Tests\Func;
+
+use DateInterval;
+use Mockery;
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\SimpleCache\CacheException;
+use Psr\SimpleCache\InvalidArgumentException;
+use wpdb;
+use WpOop\TransientCache\CachePool as TestSubject;
+use PHPUnit\Framework\TestCase;
+use Brain\Monkey\Functions;
+use function Brain\Monkey\tearDown;
+
+class CachePoolTest extends TestCase
+{
+    protected function tearDown()
+    {
+        tearDown();
+        parent::tearDown();
+    }
+
+    /**
+     * @return TestSubject&MockObject
+     */
+    public function createInstance(wpdb $wpdb, string $poolName, $defaultValue): TestSubject
+    {
+        $mock = $this->getMockBuilder(TestSubject::class)
+            ->setMethods(null)
+            ->setConstructorArgs([$wpdb, $poolName, $defaultValue])
+            ->getMock();
```

Comment:
Never mock the SUT unless it's to mock abstract methods. `CachePool` is **not** abstract. Use the constructor directly instead.

Reply:
As you can see, no method of the SUT is actually mocked. But I like to create mocks of absolutely everything. Comes handy when you wanna add expectations on other methods of the SUT, maybe with allowing original methods too.